[Jiale] iP

jialegitgud opened the pull request:
# Peter
> "Coding like poetry should be short and concise."  - Santosh Kalwar


Peter is 
- text-based
- user friendly
- FAST AND SUPER EASY TO USE!

To start using Peter, you should,
1. download it from [here](https://github.com/jialegitgud/ip/tree/master).
2. double-click the application icon.
3. add your tasks to the application.
4. and let it manage your tasks for you! 💯 

And it is completely **FREE TO USE**!

_Features:_

- [X]  Managing tasks
- [X]  Managing deadlines
- [X]  Managing events
- [ ] Providing reminders

_______________________________________________________________________________________________________________________________________

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
```
public static void main(String[] args) {
    new Peter().run();
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

jialegitgud committed:
add l0 functionalities

jialegitgud committed:
fix typo in l0

jialegitgud committed:
add l1 functionalities

jialegitgud committed:
add l2 functionalities

jialegitgud committed:
add task class

jialegitgud committed:
add l3 functionalities

jialegitgud committed:
add subclasses -task

jialegitgud committed:
complete l4 functionalities

jialegitgud committed:
added automated testing

jialegitgud committed:
added l6 functionalities

jialegitgud committed:
add l5 functionalities & Exceptions

jialegitgud committed:
modify to use enums

jialegitgud committed:
Add file IO (l7)

jialegitgud committed:
Merge pull request #1 from jialegitgud/branch-Level-7

Merge branch-Level-7 to Master (l7)

jialegitgud committed:
Refactor to use DateTime (l8)

jialegitgud committed:
Merge pull request #2 from jialegitgud/branch-Level-8

Add L8 functionalities

jialegitgud committed:
Refactor code for OOP with exception checks

jialegitgud committed:
Package files into separate packages

jialegitgud committed:
Merge branch 'add-gradle-support' of https://github.com/nus-cs2103-AY2425S1/ip into master

jialegitgud committed:
Fix gradle configurations

jialegitgud committed:
Rename packages to lowercase

jialegitgud committed:
Fix import bugs in CommandParser

jialegitgud committed:
Add storage tests and fixed storage bugs

jialegitgud committed:
Add tasklist tests and made Task abstract

jialegitgud committed:
Add find functionality (l9)

jialegitgud committed:
Merge pull request #3 from jialegitgud/branch-Level-9

Add find functionality (l9)

jialegitgud committed:
Refactor for coding conventions and clarity

jialegitgud committed:
Add JavaDocs for public functions

jialegitgud committed:
Merge branch 'A-JavaDoc' of https://github.com/jialegitgud/ip into master

jialegitgud committed:
Add JavaDoc for "Find"

jialegitgud committed:
Merge branch 'A-CodingStandard' of https://github.com/jialegitgud/ip into master

jialegitgud committed:
Fix merge error on CommandParser

jialegitgud committed:
Fix raised checkstyle issues

jialegitgud committed:
fix raised checkstyle issues

jialegitgud committed:
Merge branch 'master' of https://github.com/jialegitgud/ip into master

jialegitgud committed:
Add initial UI changes

MAOXIONGKAI commented on src/main/java/Exceptions/UnknownCommandException.java:
```diff
@@ -0,0 +1,10 @@
+package exceptions;
+
+/**
+ * Raised exception on unknown command input.
```
Just a suggestion 🤔 Would it be better to change the javadoc description from "Raised" to "Raises" as according to the [javadoc guideline in Week 3 topics](https://nus-cs2103-ay2425s1.github.io/website/schedule/week3/topics.html)

MAOXIONGKAI commented on src/main/java/Peter/Peter.java:
```diff
@@ -0,0 +1,41 @@
+public class Peter {
+    public static final String FILE_PATH = "tasks/data.txt";
+    private Storage store;
+    private TaskList tl;
```
Avoid abbreviation for variables and spell them out in full, "tasks" or "tasklist" would be a better name over here

MAOXIONGKAI commented on src/main/java/Tasks/Deadlines.java:
```diff
@@ -0,0 +1,29 @@
+public class Deadlines extends Task {
+    private LocalDateTime deadlineInfo;
+    private static final DateTimeFormatter PATTERN_DISPLAY = DateTimeFormatter.ofPattern("MMM dd yyyy HH:mm");
+    private static final DateTimeFormatter PATTERN_READ = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm");
```
I like the use of SCREAMING_SNAKE_CASE for constant over here, which makes the code make a lot more readable

MAOXIONGKAI commented on src/main/java/Utilities/Storage.java:
```diff
@@ -0,0 +1,135 @@
+                    tasks.add(new Event(splits[2], splits[3].strip(), splits[4].strip()));
+                    break;
+                default:
+                    break;
```
I like how you include a default case here for clarity and code safety purpose! 😄

MAOXIONGKAI commented on src/main/java/Utilities/Storage.java:
```diff
@@ -0,0 +1,135 @@
+            }
+            fileScanner.close();
+
+            String s = fileContents.get(index);
```
Would it better to replace "s" with something more specific? I don't know what exactly is "s" just by looking at this piece of code, maybe the future you won't know about this anymore either, so a name that can explains the variable itself is probably better?

MAOXIONGKAI reviewed (commented):
LGTM, just a few nits to fix. I like the implementation of your code, good luck on your project! 😄

jialegitgud committed:
Refactor Ui and Storage

jialegitgud committed:
Fix minor style issues

jialegitgud committed:
Complete L10 GUI functions

jialegitgud committed:
Merge pull request #4 from jialegitgud/branch-Level-10

Add l10 functionalities

sayomaki commented on src/main/java/Peter/DialogBox.java:
```diff
@@ -0,0 +1,60 @@
+     * Flips the dialog box such that the ImageView is on the left and text on the right.
+     */
+    private void flip() {
+        ObservableList<Node> tmp = FXCollections.observableArrayList(this.getChildren());
```
The variable naming is a bit ambiguous here, but good that you have added a comment to the function so that people know what it does!

sayomaki commented on src/main/java/Peter/MainWindow.java:
```diff
@@ -0,0 +1,54 @@
+import javafx.scene.image.Image;
+import javafx.scene.layout.AnchorPane;
+import javafx.scene.layout.VBox;
+/**
```
Maybe it would be nice to have a space between the imports and the javadoc?

sayomaki commented on src/main/java/Peter/Peter.java:
```diff
@@ -0,0 +1,23 @@
+    private TaskList tasklist;
+
+    public Peter() {
+        this.store = new Storage(FILE_PATH);
```
Since `store` is not ambiguous here, omitting the `this` would be okay (refer to code guidelines under "Avoid unnecessary use of this with fields.")

sayomaki commented on src/main/java/Tasks/Deadlines.java:
```diff
@@ -0,0 +1,29 @@
+
+    public Deadlines(String name, String details) {
+        super(name);
+        this.deadlineInfo = LocalDateTime.parse(details, PATTERN_READ);
```
Here the reference to the field `deadlineInfo` is not ambiguous, can possibly omit `this` keyword

sayomaki commented on src/main/java/Tasks/Task.java:
```diff
@@ -0,0 +1,39 @@
+    }
+
+    /**
+     * Toggles isDone as true.
```
I don't think the javadoc matches what the method actually does here :D

sayomaki commented on src/main/java/Tasks/Task.java:
```diff
@@ -0,0 +1,39 @@
+        return (isDone ? "X" : " ");
+    }
+
+    public abstract String getWriteTaskInfo();
```
Since this is an abstract method (which needs to be implemented by subclasses), would it be better to include some javadoc on what should the output look like? (or maybe what the method should do?)

sayomaki commented on src/main/java/Utilities/CommandParser.java:
```diff
@@ -0,0 +1,96 @@
+     * @param input String command for processing.
+     * @param tl TaskList class for maintaining tasks.
+     * @param store Storage class for writing to files.
+     * @return String value for user response.
```
Good use of javadoc annotations to describe what each parameter is and the return value of the method!

sayomaki commented on src/main/java/Utilities/Storage.java:
```diff
@@ -0,0 +1,135 @@
+            // Admin Message for Generation
+            System.out.println("File has been generated.");
+            try {
+                this.file.createNewFile();
```
`this` here can be omitted as the reference is clear.

sayomaki commented on src/main/java/Utilities/UI.java:
```diff
@@ -0,0 +1,52 @@
+import tasks.Task;
+
+// Handles all system interactions
+public class Ui {
```
Maybe you can add some javadoc for the public methods of this class? :)

sayomaki commented on src/test/java/utilities/TaskListTest.java:
```diff
@@ -0,0 +1,33 @@
+    private static final String ACTUAL_STRING = "1.[T][ ]  borrow book \n"
+            + "2.[E][X]  project meeting from: Dec 29 2024 12:00 to: Dec 29 2024 17:00 \n"
+            + "3.[T][ ]  borrow notes";
+    private Storage store = new Storage(FILE_PATH);
```
I think there can be a space above this line as this is separate from the 2 constants above.

sayomaki reviewed (commented):
Overall coding standard mostly complies with the coding guidelines, but there are some minor issues (see comments below!)

jialegitgud committed:
Add JAR File

jialegitgud committed:
Fix general code quality

There are several style fixes for the methods within the code.

Added JavaDocs for clarity of the UI class, and fixed the JavaDocs
for several classes with inaccurate representations.

Other minor fixes include removing magic strings in the CommandParser
class, and adding spaces for greater readibility where required.

jialegitgud committed:
Merge pull request #5 from jialegitgud/branch-A-CodeQuality

Merge: Fix general code quality

jialegitgud committed:
Add assertion checking in logic

jialegitgud committed:
Merge branch 'master' of https://github.com/jialegitgud/ip into master

jialegitgud committed:
Merge branch 'master' of https://github.com/jialegitgud/ip into master

jialegitgud committed:
Add Tagging Functionalities

There is currently no way to group similar tasks together, and the task
list remains sorted only based on order of addition.

There must be a method to group tasks so the find command can
be put to better use.

Tagging functionality has been added so that similar tasks can be
found together in the same list by searching the tag.

This is preferable so that different tasks added at different times but
are related can be viewed together.

jialegitgud committed:
Merge pull request #6 from jialegitgud/branch-C-Tagging

Add Tagging

jialegitgud committed:
Rename UI.java to Ui.java

jialegitgud committed:
Add Enum for code clarity

jialegitgud committed:
Merge branch 'master' of https://github.com/jialegitgud/ip into branch-A-CodeQuality

jialegitgud committed:
Merge pull request #7 from jialegitgud/branch-A-CodeQuality

Add Enum for Commands

jialegitgud committed:
Refactor to extract Task Creation

jialegitgud committed:
Add Streams for TaskList

jialegitgud committed:
Refactor for missing JavaDocs and code quality

jialegitgud committed:
Add "BRO" personality to Peter

jialegitgud committed:
Fix UI spacing

jialegitgud committed:
Fix style errors and visual bugs

jialegitgud committed:
Fix test issues

jialegitgud committed:
Add user guide

jialegitgud committed:
Fix path errors

jialegitgud committed:
Fix missing dependencies
